### PR TITLE
CH-289: Implementation of Alert occurrences page as per the new design

### PR DESCRIPTION
**Pull request by NareshGopal**
# Problem Statement
- Modify the "Alert occurrences" page as per the [new design](https://www.figma.com/file/HuYNdGvVazilCkD8mJO1qQ/LR-Alert-page-redesign-options?node-id=1094%3A68101)

# Design
-  [New design](https://www.figma.com/file/HuYNdGvVazilCkD8mJO1qQ/LR-Alert-page-redesign-options?node-id=1094%3A68101)

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: manager/gui/src/lib/components/SgtAdvanceSearch/SgtAdvanceSearch.vue [hunk 1]
```diff
@@ -76,6 +76,7 @@
                     v-model="row.value"
                     outlined
                     dense
+                    :disabled="row.required"
```
Comment (sujit-h-singh) [inline manager/gui/src/lib/components/SgtAdvanceSearch/SgtAdvanceSearch.vue:79]: disabled is not directly proportional with required

File: manager/gui/src/components/alerts/LrAlert.vue [hunk 2]
```diff
@@ -76,33 +83,31 @@ import { SgtFilterObject } from "@/lib/components/SgtChips/SgtFilterObject.model
 import SgtDialog from "@/lib/components/SgtDialog/SgtDialog.vue";
 import { SgtDialogModel } from "@/lib/components/SgtDialog/SgtDialog.model";
 import { create } from "vue-modal-dialogs";
+import AlertMixin from "@/mixins/AlertMixin";
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlert.vue:86]: Any particular reason to use Mixins?

File: manager/gui/tests/unit/alert/LrAlertDetails.spec.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    expect(mockGetData).toHaveBeenCalled();
+
+    const alertTitle = wrapper.find(".top-section .section-title");
+    expect(alertTitle.html()).toContain("Alert");
+    const occurrencesTitle = wrapper.find(".alert-title-container");
+    expect(occurrencesTitle.html()).toContain("Occurrences");
+  });
+
+  it("Checks whether the alert details section contains all the titles.", async () => {
+    const alertDetailsContainer = wrapper.find(".alert-details-container");
```
Comment (sujit-h-singh) [inline manager/gui/tests/unit/alert/LrAlertDetails.spec.ts:119]: Need to test data as well for alert details.

File: manager/gui/src/components/alerts/LrAlert.vue [hunk 4]
```diff
@@ -121,23 +126,16 @@ export default class LrAlert extends Vue {
...
-          });
-        }
-        return element;
-      });
-      this.alertConst.searchConfig.advanceForm = updatedAdvanceForm;
-    }
+  setFilter(filterName: string, filterValue: string) {
+    const advanceForm = [...this.alertConst.searchConfig.advanceForm];
+    const updatedAdvanceForm = advanceForm.map((element) => {
+      if (element.name === filterName) {
```
Comment (sujit-h-singh) [inline manager/gui/src/components/alerts/LrAlert.vue:132]: need to modify this for non-editable field, and also use find rather than map.

**Review by sujit-h-singh — CHANGES_REQUESTED** [review]
Need additional test cases for LrAlertInformation.vue

File: manager/gui/src/lib/components/SgtAdvanceSearch/SgtAdvanceSearch.vue (hunk 1 above)
Comment (NareshGopal) [reply manager/gui/src/lib/components/SgtAdvanceSearch/SgtAdvanceSearch.vue]: Added a new property 'editable' to handle disabling inputs.

File: manager/gui/src/components/alerts/LrAlert.vue (hunk 2 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.vue:86]: Justification has been given over the call. The same functionality is required on all alerts screen and alert occurrence screen.

File: manager/gui/src/components/alerts/LrAlert.vue (hunk 4 above)
Comment (NareshGopal) [reply manager/gui/src/components/alerts/LrAlert.vue]: Added a new property 'editable' to handle disabling inputs. 'map' function is replaced with 'findIndex'.

File: manager/gui/src/lib/components/SgtChips/SgtChips.vue [hunk 5]
```diff
@@ -22,7 +22,7 @@
         dark
         v-for="(chip, i) in chips"
         :key="i"
-        :close="chip.required ? false : true"
+        :close="chip.editable"
```
Comment (sujit-h-singh) [inline manager/gui/src/lib/components/SgtChips/SgtChips.vue:25]: !chip.required

**Review by sujit-h-singh — APPROVED** [review]

**Pull request merged**